### PR TITLE
WIP Enav version - move S104/S111 logic into S100 class

### PR DESCRIPTION
**Pull request by princessmittens**
* Added types to params in S104, S100 and S111 classes
* Added _populate_group_metadata function in S100 
* Added _create_positioning_group function in S100
* Moved logic for _update_feature_metadata into S100 class 
* Expanded on init functions in S111/S104 to have hardcoded values that are custom to each class (passed into S100 constructor)
* Added s100_util file for utility methods and an attribute class 

Note that there is a descrepancy between my branch and the eastBoundLatitude, could you take a look at this? I'm not sure why this is the case but I thought maybe you pushed a fix for it while I was working on the branch.

### CODE REVIEW

**Comment by princessmittens** [comment]
No, I don't think it's a thing in github. Different I guess!

**Comment by glmcr** [comment]
> No, I don't think it's a thing in github. Different I guess!

Not sure but I think that we have to use the "convert to draft" button for getting the same gitlab "WIP" thing effect

**Review by maximecarre1 — CHANGES_REQUESTED** [review]
Print statements (new and old) are causing problems when generating the open api yml file, they should be changed to generate entries in the log file.

File: config.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 From cache: True
-Created: 2022-08-11 18:31:54.311503
-Expires: 2022-08-12 18:31:54.311487
+Created: 2022-08-18 18:50:30.023440
+Expires: 2022-08-19 18:50:30.023402
```
[the remark this answers is not in the document]
Comment (princessmittens) [reply config.yml]: I have made the changes, can confirm this is correct? Thanks!

[the remark this answers is not in the document]
Comment (maximecarre1) [reply config.yml]: Yes, everything looks fine for now. We are going to replace the config file later anyway to deal with [issue 11 ](https://github.com/DFO-CHS-Dynamic-Hydrographic-Products/IWLS_pygeoapi/issues/11)

**Comment by princessmittens** [comment]
I think this is ready to merge if the changes have been reviewed?

**Pull request merged**